app/src/main/java/com/arctouch/test/ui/activity/ArcTouchActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.arctouch.test.ui.activity
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import com.arctouch.test.di.Injectable
+import dagger.android.AndroidInjection
+
+abstract class ArcTouchActivity : AppCompatActivity(), Injectable {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        AndroidInjection.inject(this)
+        setUp(savedInstanceState)
+    }
+
+    override fun onPostCreate(savedInstanceState: Bundle?) {
+        super.onPostCreate(savedInstanceState)
+        onViewCreated(savedInstanceState)
+        bind(savedInstanceState)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                finish()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    internal open fun setUp(savedInstanceState: Bundle?) {
+
```

Comment:
If you're not using these methods, they could be removed.

Reply:
This is a BaseActivity, these methods are override in another activities.

Reply:
Ok, but why did not you use the Activity's methods `onCreate`, `onPostCreate` directly and remove those empty methods?

Reply:
Well.. I'm used to do that just for organization purpose. As I'm used to work in a lot of projects with a lot of novice people I let theses methods organized to make other developers bind view model after set up the views. I agree that looking at it right now they can be avoided, specially the setUp method.